src/Symfony/Component/Messenger/Transport/SqsExt/Connection.php

```diff
@@ -0,0 +1,362 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\SqsExt;
+
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\Messenger\Exception\InvalidArgumentException;
+use Symfony\Component\Messenger\Exception\TransportException;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * A SQS connection.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ *
+ * @internal
+ * @final
```

Comment:
Why not use `final class Connection` instead?

Reply:
just for consistency with `AmqpExt\Connection`, `Doctrine\Connection`, `RedisExt\Connection` in this same component

Reply:
I would make them `final` except if their is a good reason.

Reply:
done

Reply:
Now I remember why.

> Class "Symfony\Component\Messenger\Transport\SqsExt\Connection" is declared "final" and cannot be mocked

Reply:
> Now I remember why.
> 
> > Class "Symfony\Component\Messenger\Transport\SqsExt\Connection" is declared "final" and cannot be mocked

This means => `Please mock interface instead of concrete classes` ^^

Reply:
not sure an internal final service needs an interface

Reply:
Would suggest not to mock this.. isn't this possible?